### ISSUE TITLE
improve screen reader context and navigation of Cell outputs

### ISSUE THREAD

**Issue by amunger** [microsoft/vscode#176292]
Tabbing to a cell output is missing context about which cell it is for.

The suggestion from @jooyoungseo is adding markdown headers for each cell output to describe what the output is for and provide a clearer boundary around the output.
This would also allow screen readers to iterate over outputs since they can navigate directly to markdown headers.

**Comment by amunger** [comment]
some complications with the output webview

- focusing the output will first read the previously focused output line, then read the currently focused one right after
- the order of the output cells in the DOM does not match the order on the screen. I think they are just appended on when executing, but you could also rearrange the cells after executing.
  - the result is that browsing in the output webview seems to jump around to random cells
  - this sounds pretty hard to fix to actually keep them consistent, but I imagine it would be pretty confusing for the user
- The cell output doesn't have context on which cell it's for, which is why that PR just has a generic 'begin cell output' label
  - probably easier to fix, but I haven't fully thought it through

But, because there is currently a potentially helpful initiative for a new tool from @meganrogge that would go in an entirely different direction, we will hold off on attempting further fixes for this until we know more.

**Comment by dennischenfeng** [comment]
Just chiming in to voice my support for this ticket, especially about cell output labeling and boundaries.
One use case that makes notebooks hard to use for me is that I can't quickly arrow down to the bottom of a particular cell output's stack trace error message, since there are no boundaries between cell outputs..

**Comment by amunger** [comment]
thanks for the feedback @dennischenfeng - we should have some good improvements for cell output from https://github.com/microsoft/vscode/issues/183208, in insiders now and in the next stable release next week.
I'm going to close this as a duplicate of that one, but feel free to leave feedback there if you try it out, or open a new issue.
